app/celer-g4/DetectorConstruction.cc

```diff
@@ -319,7 +319,7 @@ void DetectorConstruction::foreach_detector(F&& apply_to_range) const
             ++stop;
         } while (stop != detectors_.end() && start->first == stop->first);
 
-        apply_to_range(start, stop);
+        std::forward<F>(apply_to_range)(start, stop);
```

Comment:
Since we're applying this multiple times, I don't think we want to forward it

Reply:
Do we need the forwarding reference then?

Reply:
If F is a class with mutable operator() I don't think constructing ref will work?